app/models/job.rb

```diff
@@ -12,6 +12,11 @@ class Job < ActiveRecord::Base
   validate :validate_globally_unlocked
 
   ACTIVE_STATUSES = %w[pending running cancelling].freeze
+  VALID_STATUSES = ACTIVE_STATUSES + %w[failed errored succeeded cancelled].freeze
+
+  def self.valid_status?(status)
```

Comment:
feel like this should live in the model, since it should know.  this is also a pattern that's in use for a couple of other things.